libs/auth/src/interfaces/auth-facade.interface.ts

```diff
@@ -0,0 +1,11 @@
+import { Observable } from 'rxjs';
+import { Action } from '@ngrx/store';
+
+import { DaffStoreFacade } from '@daffodil/core';
+
+export interface DaffAuthFacadeInterface extends DaffStoreFacade<Action> {
```

Comment:
This should be generic, because it'll need have a property called `auth$: Observable<T>` where T extends DaffAuth. That way the facade is completely reusable in the case that the app dev provides their own model that extends `DaffAuth` and transformer and such. But it's probably a good idea to keep `token$`, since that won't change

Reply:
Honestly, I don't think we even need an interface or token for the facade, since the token won't be referenced anywhere in daffodil, and if someone uses their own facade, there's really no reason for them to follow the pattern we've used. They should really never need to use anything but the facade we provide anyway

Reply:
The injection token is useful for the mock facade in the testing section. The interface might still be useful for typing purposes.

Reply:
In the testing section, you can just do 
```
providers: [
  { provide: DaffAuthFacade, useClass: MockDaffAuthFacade }
]
```
so the injection token isn't needed for that. Unless there's a particular instance you're thinking of where the interface can be useful, I'd say YAGNI it, because I can't think of anytime when it would be used.

Reply:
I agree with @griest024 about the interface. If we're going to provide a mock facade, we should be providing an interface to ensure that these remain compatible over time.

Reply:
I erroneously believed that I had to define the mock facade before the real facade.